model/channel.go

```diff
@@ -131,12 +131,12 @@ func (o *Channel) IsValid() *AppError {
 		return NewAppError("Channel.IsValid", "model.channel.is_valid.display_name.app_error", nil, "id="+o.Id, http.StatusBadRequest)
 	}
 
-	if !IsValidChannelIdentifier(o.Name) {
-		return NewAppError("Channel.IsValid", "model.channel.is_valid.2_or_more.app_error", nil, "id="+o.Id, http.StatusBadRequest)
+	if (o.Type == CHANNEL_OPEN || o.Type == CHANNEL_PRIVATE) && !IsValidChannelIdentifier(o.Name) {
```

Comment:
I have to be more explicit about how to validate each channel because the direct and group channels doesn't follow exactly the same rules, so they should be validated in the current way, and only the public and private channels get validated with the new restrictions.